cupy/lib/polynomial.pyx

```diff
@@ -10,7 +10,7 @@ cimport cython  # NOQA
 
 @cython.profile(False)
 cdef inline _should_use_rop(x, y):
-    # case: python scalar + poly1d
+    # case: python scalar ± poly1d
     if cupy.isscalar(x) and isinstance(y, poly1d):
         return False
```

Comment:
How about raising `TypeError` here?

Reply:
Hmm, why?

Reply:
Sorry, I was confused with numpy scalar case.

Reply:
Anyway, I think these lines can be removed because L18 always returns `False` in this case.